src/Cryptol/TypeCheck/Solver/InfNat.hs

```diff
@@ -122,23 +122,22 @@ nMod (Nat x) (Nat y)  = Just (Nat (mod x y))
 nMod (Nat x) Inf      = Just (Nat x)          -- inf * 0 + x = 0 + x
 
 -- | @nCeilDiv msgLen blockSize@ computes the least @n@ such that
--- @msgLen <= blockSize * n@. It is undefined when @blockSize = 0@.
--- It is also undefined when either input is infinite; perhaps this
--- could be relaxed later.
+-- @msgLen <= blockSize * n@. It is undefined when @blockSize = 0@,
+-- or when @blockSize = inf@. @inf@ divided by any positive
+-- finite value is @inf@.
 nCeilDiv :: Nat' -> Nat' -> Maybe Nat'
+nCeilDiv _       Inf      = Nothing
 nCeilDiv _       (Nat 0)  = Nothing
-nCeilDiv Inf     _        = Nothing
-nCeilDiv (Nat _) Inf      = Nothing
+nCeilDiv Inf     (Nat _)  = Just Inf
```

Comment:
You probably need to update `lib/CryptolTC.z3` to match the new behavior

Reply:
Also, update `TypeCheck/Solver/Numeric/Fin.hs` as it is currently assume that the result is always `fin`

Reply:
Any (maybe) `src/Cryptol/TypeCheck/Solver/Numeric/Interval.hs`

Reply:
Do you have a sense about how and if the interval arithmetic needs to change?

Reply:
Not sure, but I was thinking that since previously the answer was always finite, and now it can be infinite sometimes, we may have to tweak something to make a bit larger

Reply:
The interval computations fall back on `nCeilDiv` for the new cases that matter.  After staring at the interval code for awhile, I'm pretty confident they don't need changing.